app/src/main/java/com/example/sun/data/repository/source/local/CurrentWeatherLocalDataSource.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.sun.data.repository.source.remote
+
+import com.example.sun.data.model.CurrentWeather
+import com.example.sun.data.repository.source.CurrentWeatherDataSource
+
+class CurrentWeatherLocalDataSource : CurrentWeatherDataSource.Local {
+    override fun getCurrentWeatherLocal(listener: OnResultListener<MutableList<CurrentWeather>>) {
```

Comment:
Use List instead of MutableList
```suggestion
    override fun getCurrentWeatherLocal(listener: OnResultListener<List<CurrentWeather>>) {
```